mbs_results/utilities/csw_to_spp_converter.py

```diff
@@ -251,3 +253,33 @@ def load_and_join_finalsel(
     finalsel_data["formtype"] = "0" + finalsel_data["formtype"].astype(str)
     finalsel_data.rename(columns=finalsel_column_remapper, inplace=True)
     return df.merge(finalsel_data, on=["reference", "period"], how="left")
+
+def validate_nil_markers(df: pd.DataFrame, log_file: str) -> pd.DataFrame:
```

Comment:
this function needs to be called in the top level function, `create_snapshot()`, perhaps on the `qv_df` before it becomes `responses`